### PR TITLE
added box detached api

### PR DESCRIPTION
**Pull request by artenator**
hey again @lvh ! 

This fixes issue #5 

Let me know what you think.

### CODE REVIEW

File: src/caesium/crypto/box.clj [hunk 1]
```diff
@@ -253,3 +253,38 @@
...
+        mac (bb/alloc macbytes)]
+    (box-detached-to-bufs!
+     c
+     mac
+     (bb/->indirect-byte-buf ptext)
+     (bb/buflen ptext)
+     (bb/->indirect-byte-buf nonce)
+     (bb/->indirect-byte-buf pk)
+     (bb/->indirect-byte-buf sk))
+    {:ctext (bb/->bytes c)
```
Comment (lvh) [inline src/caesium/crypto/box.clj]: this should probably use the same name as the upstream api: `:c`

**Comment by lvh** [comment]
this looks great except for the name nit; thanks!

**Comment by artenator** [comment]
@lvh 

Done!

Thank you. I am learning a lot

**Comment by lvh** [comment]
Neat, thanks! I should really cut a release for this :)

**Pull request merged**